src/server/frontend_wayland/wl_pointer.cpp

```diff
@@ -321,11 +321,28 @@ void mf::WlPointer::maybe_frame()
 
 namespace
 {
+struct CursorSurfaceRole : mf::NullWlSurfaceRole
+{
+    mf::WlSurface* const surface;
+    mf::CommitHandler* const commit_handler;
+    explicit CursorSurfaceRole(mf::WlSurface* surface, mf::CommitHandler* commit_handler) :
+        NullWlSurfaceRole(surface),
+        surface{surface},
+        commit_handler{commit_handler} {}
+
+    void commit(mir::frontend::WlSurfaceState const& state) override
```

Comment:
If `this` is destroyed before the surface, the role is not cleared and `commit()` may be called on a freed object.

Reply:
Resolved by my commit